infra/Pos/Slotting/MemState/Class.hs

```diff
@@ -8,24 +8,52 @@ import           Control.Monad.Trans (MonadTrans)
 import           Universum
 
 import           Pos.Core.Types      (EpochIndex, Timestamp)
-import           Pos.Slotting.Types  (SlottingData)
+import           Pos.Slotting.Types  (EpochSlottingData)
+
 
 -- | 'MonadSlotsData' provides access to data necessary for slotting to work.
+-- Both _current_ and _next_ epoch @SlottingData@ need to be present since it's impossible
+-- to create @SlottingData@ without them.
 class Monad m => MonadSlotsData m where
 
-    getSystemStart :: m Timestamp
+    getSystemStartM :: m Timestamp
+
+    getAllEpochIndicesM :: m [EpochIndex]
+
+    getCurrentEpochIndexM :: m EpochIndex
+
+    getNextEpochIndexM :: m EpochIndex
```

Comment:
Ok, I guess I can move all the associated functions to `Functions`, now they are mostly in `Types`. Still, typeclass + datatypes seems cleaner than typeclass + datatypes + `Types` + `Functions` + ...
I guess I'm used to looking at the code through OOP :eye: